cmd/libsnap-confine-private/utils.h

```diff
@@ -20,6 +20,17 @@
 #include <stdbool.h>
 #include <stdlib.h>
 
+/**
+ * Macro which calculates array size.
+ *
+ * Based on ARRAY_SIZE from the Linux kernel, see
+ * https://elixir.bootlin.com/linux/v6.13.3/source/include/linux/array_size.h#L11
+ */
+#define SC_ARRAY_SIZE(arr)                                                                                  \
+    (sizeof(arr) / sizeof((arr)[0]) + ((int)sizeof(struct {                                                 \
+         _Static_assert(!__builtin_types_compatible_p(typeof(arr), typeof(&(arr)[0])), "must be an array"); \
```

Comment:
`_Static_assert` which is awesome actually becomes a part of the C23 standard just as a random fact!